### ISSUE TITLE
RKObjectMapping 0.22.0 NULL serialization by default

### ISSUE THREAD

**Issue by jerez** [RestKit/RestKit#1714]
After updating from 0.21 to 0.22 my `PATCH` requests stopped working properly, all `nil` values are serialized to null by default, even setting `assignsDefaultValueForMissingAttributes` to `NO` Explicitly

Ex: some object with a property1 set to `any` and property2 set to `nil` results in:

`0.21.0` -> `{"property1":"any"}`

`0.22.0` -> `{"property1":"any", "property2":null}`

**Comment by biddster** [comment]
Agreed, this is causing me a problem too when I'm talking to a .NET rest service.

**Comment by biddster** [comment]
I've submitted PR https://github.com/RestKit/RestKit/pull/2157 with a proposed fix.
